src/System.Management.Automation/engine/MshCommandRuntime.cs

```diff
@@ -2991,21 +3084,33 @@ internal ActionPreference VerbosePreference
 
                 if (!_isVerbosePreferenceCached)
                 {
-                    bool defaultUsed = false;
-                    _verbosePreference = Context.GetEnumPreference<ActionPreference>(
-                        SpecialVariables.VerbosePreferenceVarPath,
-                        _verbosePreference,
-                        out defaultUsed);
+                    _verbosePreference = Context.GetEnumPreference<ActionPreference>(SpecialVariables.VerbosePreferenceVarPath, _verbosePreference, out _);
+
+                    // If the host couldn't prompt for the verbose action anyways, change it to 'Continue'.
+                    // This lets hosts still see debug output without having to implement the prompting logic.
+                    if ((CBhost.ExternalHost.UI == null) && (_verbosePreference == ActionPreference.Continue))
+                    {
+                        _verbosePreference = ActionPreference.Continue;
+                    }
+
+                    _isVerbosePreferenceCached = true;
```

Comment:
FYI for reviewers, lines 3089-3096 were added because it seemed that they were missing (we were checking if the verbose preference was cached or not, but never actually marking it as cached).

Reply:
See other comments related to why the lack of caching is actually a bug for Warning and Verbose messages.